File: 2025/09 September/25 Generate Binary Numbers/main.cpp
// Link: https://www.geeksforgeeks.org/problems/generate-binary-numbers-1587115620/1

#include <iostream>
#include <vector>
#include <queue>
using namespace std;

class Solution
{
public:
    vector<string> generateBinary(int n)
    {
        vector<string> res;
        queue<string> q;

        q.push("1");

        while (n--)
        {
            string s1 = q.front();
            q.pop();
            res.push_back(s1);

            string s2 = s1;

            q.push(s1.append("0"));

            q.push(s2.append("1"));
        }
        return res;
    }
};

int main()
{
    Solution obj;
    int n = 4;
    vector<string> ans = obj.generateBinary(n);
    cout << "Binary numbers from 1 to " << n << " are: ";
    for (auto i : ans)
    {
        cout << i << " ";
    }
    cout << endl;

    n = 6;
    ans = obj.generateBinary(n);
    cout << "Binary numbers from 1 to " << n << " are: ";
    for (auto i : ans)
    {
        cout << i << " ";
    }
    cout << endl;
    return 0;
}

File: 2025/09 September/25 Generate Binary Numbers/main.py
# Link: https://www.geeksforgeeks.org/problems/generate-binary-numbers-1587115620/1

from collections import deque

class Solution:
    def generateBinary(self, n):
        res = []
        q = deque()

        q.append("1")

        while n > 0:
            s1 = q.popleft()
            res.append(s1)

            s2 = s1

            q.append(s1 + "0")

            q.append(s2 + "1")

            n -= 1

        return res


if __name__ == "__main__":
    obj = Solution()
    n = 4
    result = obj.generateBinary(n)
    print(f"Binary numbers from 1 to {n}: {result}")

    n = 6
    result = obj.generateBinary(n)
    print(f"Binary numbers from 1 to {n}: {result}")
